api/src/main/resources/moduleApplicationContext.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- Beans to add to the current Application context definition -->
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+        http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-3.0.xsd">
+
+    <context:component-scan base-package="org.bahmni.module"/>
```

Comment:
This MUST not be done. Please do a scan only for your relevant package 
"org.bahmni.events" or even more specific. 
Also, I noticed that you do not have segregation on package name for omod and api .. 
for example we typically do this as "org.bahmni.events.api" and then its best to do component scan as to that package only. 
similarly, only scan relevant packages in "omod" module

Reply:
We don't have anything in omod. And i have changed component scan to org.bahmni.module.events

Reply:
Now changed to org.bahmni.module.events.api